glusterapi/common.py

```diff
@@ -1,33 +1,67 @@
-import json
-from exceptions import GlusterApiError
+import datetime
+import hashlib
 
+import jwt
 import requests
 
+from exceptions import GlusterApiError
+
 
 class BaseAPI(object):
-    def __init__(self, host, https=False, port=24007,
-                 username=None, secret=None):
-        self.host = host
-        self.port = port
-        self.scheme = "https" if https else "http"
-        self.base_url = "%s://%s:%d" % (self.scheme, self.host, self.port)
+    def __init__(self, endpoint='http://127.0.0.1:24007', user=None, secret=None, verify=False):
+        self.base_url = endpoint
+        self.user = user
+        self.secret = secret
+        self.verify = verify
+
+    def _set_token_in_header(self, method, uri, headers=None):
```

Comment:
additional check required to see `self.user is not None` and `self.secret is not None`

Reply:
done

Reply:
@Madhu-1 you may have accidentally modified the wrong commit. You should probably split the change here to common.py out of this change to requirements.txt. Then you can squash the modifications to common.py to the appropriate change.

Reply:
@phlogistonjohn  Thanks, I did git commit with amend so the changes went to the wrong commit.
updated the commits.